torch/distributed/_composable/fsdp/_fsdp_param_group.py

```diff
@@ -270,6 +270,7 @@ def reshard(self):
                 return
         self._to_sharded()
 
+    @torch.compiler.disable()
```

Comment:
To-do: separate this out and do not include in this PR
(however, we need this to compile with float8)

Reply:
will be landing ``@torch.compiler.disable_if_config_true("skip_fsdp_hooks")`` this week https://github.com/pytorch/pytorch/pull/123021